### PR TITLE
Managed queues

### PR DESCRIPTION
**Pull request by smacfarlane**
This adds the ability to add requirements to processing steps.   

Requirements will be used in place of the queue field  for mapping workers to jobs

### CODE REVIEW

File: app/helpers/steps_helper.rb [hunk 1]
```diff
@@ -1,2 +1,7 @@
 module StepsHelper
+  def available_sidekiq_queues
+    queues = Set.new(Sidekiq::Queue.all.map(&:name))
+    queues << 'default'
```
Comment (teknofire) [inline app/helpers/steps_helper.rb]: Should this have a test to only add it if sidekiq didn't include it already?  I know in the paws queue sidekiq was giving me back the default queue.

File: test/models/step_test.rb [hunk 2]
```diff
@@ -14,7 +14,7 @@ def setup
   def test_build_job_chain
     job = @step.build_job_chain(passes(:snpp))
     job.save
-    step_names = %w(SnppAcquisition SnppRtstps ViirsSdr ViirsEdr ViirsAwips ViirsGeotiff)
+    step_names = %w(SnppAcquisition SnppRtstps ViirsSdr ViirsEdr ViirsAwips)
```
Comment (teknofire) [inline test/models/step_test.rb]: just confirming but no longer testing for ViirsGeotiff in the chain?

Comment (smacfarlane) [inline test/models/step_test.rb]: Correct.  ViirsGeotiff has the enabled attribute set to false, so it should not be included in the job list.   

Maybe should add an assert/new test that specifically calls that out to make it clear?

File: app/helpers/steps_helper.rb (hunk 1 above)
Comment (smacfarlane) [inline app/helpers/steps_helper.rb]: queues is a Set, so adding in does nothing if it already exists.

File: test/models/step_test.rb (hunk 2 above)
Comment (teknofire) [inline test/models/step_test.rb]: Could probably just change the var name to enabled_step_names

**Comment by teknofire** [comment]
:+1:

**Pull request merged**